### ISSUE TITLE
Thread update

### ISSUE THREAD

**Issue by Pierrotd** [FreshRSS/FreshRSS#798]
Hi,
In Tiny Tiny RSS, updated threads come back in the principal stream as non read.
Is it be possible to add this option in FreshRSS?
Thanks

**Comment by Alkarex** [comment]
That could be possible. We would have to change a bit our approach to determinate the unicity of an article.
There could two alternative approaches at least:
1. Mark the old article as unread, and change its content with the new version;
2. or make a new article.

**Comment by Alkarex** [comment]
To test the second alternative, could can try to change to `$item->get_id(true),`
https://github.com/FreshRSS/FreshRSS/blob/beta/app/Models/Feed.php#L302
(Not tested, so proceed with caution)

**Comment by Alkarex** [comment]
Everybody is welcome to comment/vote the two possible options above regarding possible behaviours when feed entries are modified.

**Comment by Alwaysin** [comment]
I am in favor of the first option; I don't want "duplicating" articles.
But we could have a threaded view of an article with the different versions able to be reviewed.

Else, the first option seems good, but we should add an indicator saying the article has been updated (icon, text added in front of the title of the article, or whatever).

**Comment by Pierrotd** [comment]
I think the first option is the best (Mark the old article as unread, and change its content with the new version)

**Comment by Alkarex** [comment]
Example of feed updating old entries: 01Net http://rss.feedsportal.com/c/629/f/502199/index.rss
It looks like articles such as http://www.01net.com/editorial/624966/applis-android-iphone-ipad-les-bons-plans/ and http://www.01net.com/editorial/638020/01net-evening-les-5-infos-quil-ne-fallait-pas-rater-aujourdhui/ keep their GUID and are regularly updated.

Another example at OVH http://travaux.ovh.net/rss.php

**Comment by Alkarex** [comment]
I have started the work on that and should have a patch soon. I need to address other pending issues at the same time, e.g. a better way to purge old articles, because now we have old articles that may get updated, and because our old approach was not robust enough anyway.
https://github.com/FreshRSS/FreshRSS/issues/493
https://github.com/FreshRSS/FreshRSS/blob/e2da6e6e6b871dc3dbc289cdd40ba401a21d8e91/app/Controllers/feedController.php#L335

**Comment by Alkarex** [comment]
Example of problematic feed for change detection, which makes irrelevant random hidden changes to the content of items: LeMonde.fr http://rss.lemonde.fr/c/205/f/3050/index.rss

``` xml
<enclosure url="http://s2.lemde.fr/image/XXX.jpg" length="776477" type="image/jpeg" />
```

The image server randomly changes between `s1.lemde.fr` and `s2.lemde.fr`...
We will need an option to disable change detection, in particular for such feeds with bad behaviour.

**Comment by Alkarex** [comment]
@Pierrotd Could you be interested in testing the new feature you asked?
In the [/dev branch](https://github.com/FreshRSS/FreshRSS/tree/dev), there is now a user option to specify that updated articles should be marked unread:

![mark updated](https://cloud.githubusercontent.com/assets/1008324/7554498/59063102-f72b-11e4-9499-fce6420db87a.png)

So far, there is only a global option per user, but I am considering adding a custom option per feed, so that each user can have a different policy for each feed.

**Comment by Pierrotd** [comment]
I will try as soon as I can. I hope before the end of the week. I will tell you when I install the dev version.
Good idea to add the option per feed.

**Comment by Pierrotd** [comment]
I use MariaDB. The update of read  articles works. I compared the same feed in tiny tiny rss and the dev version of FreshRSS.

**Comment by Alkarex** [comment]
@Pierrotd Thanks for the feedback :-)

**Comment by Alkarex** [comment]
To address @Alwaysin comment, a possibility to keep the information that an article has been updated since it was read would be to use the SQL field `entry.is_read`, which we currently use as {0:unread, 1:read}, but which we could use without having to change the database as {-1:updated,0:unread,1:read}. Only a few changes would have to be made, mainly replacing `==0` by `<=0` for checking the unread state. The SQL indexes should still be almost as good.
P.S. And there will be the question of how to present this information in the Web interface.

**Comment by Alwaysin** [comment]
How can one check if an article has really been updated by this mechanism, instead of a potential bug?

Following https://github.com/FreshRSS/FreshRSS/issues/493#issuecomment-102491457, I have so-called updates coming, but without apparent change. The date is also the same.

Exemple with http://torrentfreak.com/ and http://www.nextinpact.com feeds (the latter usually specify [MAJ] in the title when it is happening).

I think it may be a bad behaviour as described concerning http://www.lemonde.fr (https://github.com/FreshRSS/FreshRSS/issues/798#issuecomment-89655370) but I don't really know how to check.

**Comment by Alkarex** [comment]
@Alwaysin A low-tech approach to debugging is to manually save copies of the feed and do a diff (I think this is the easiest and most robust). In the logs, there is also an MD5 hash before/after the update.

**Comment by Alkarex** [comment]
The changes so far will be included in the upcoming FreshRSS version 1.1.1-beta, and the remaining improvements will target the following 1.1.2-beta

**Comment by Alkarex** [comment]
The work continues in https://github.com/FreshRSS/FreshRSS/issues/891 for the control per feed.
